File: DFS/DFS.js
class Graph{
    constructor(V){
        this.V = V
        this.adj = {}
        for(let i = 0; i < V; i++){
            this.adj[i] = []
        }
    }

    addEdge(src, dest){
        this.adj[src].push(dest)
    }

    printGraph(){
        for (let i = 0; i < this.V; i++){
            let output = `\n Vertex ${i}:`
            for(let x = 0; x < this.adj[i].length; x++){
                output += `-> ${this.adj[i][x]}`
            }
            console.log(output)
        }
    }

    DFS(start){
        let visited = []
        for(let i = 0; i < this.V; i++)
            visited.push(false)
        let stack = []
        stack.push(start)
        let output = `\n DFS: `

        while(stack.length){
            start = stack.pop()

            if(!visited[start]){
                output += `${start}-> `
                visited[start] = true
            }

            for (let i = 0; i < this.adj[start].length; i++){
                if(!visited[this.adj[start][i]])
                    stack.push(this.adj[start][i])
            }
        }
        console.log(output)
    }
}

let adj = new Graph(5)
adj.addEdge(0, 1)
adj.addEdge(0, 2)
adj.addEdge(0, 3)
adj.addEdge(1, 2)
adj.addEdge(3, 4)
adj.printGraph()
adj.DFS(0)

File: DFS/DFS.py
class Graph:
    def __init__(self, V):
        self.V = V
        self.adj = [[] for i in range(V)]
    
    def addEdge(self, src, dest):
        self.adj[src].append(dest)
    
    def printGraph(self):
        for i in range(self.V):
            print(f"\n Vertex {i}:", end='')
            for x in self.adj[i]:
                print(f"-> {x}", end='')
            print(f'\n', end='')

    def DFS(self, start):
        visited = [False for i in range(self.V)]
        stack = []
        stack.append(start)

        print(f"\n DFS: ", end='')

        while (len(stack)):
            start = stack.pop()
            
            if (not visited[start]):
                print(f'{start}-> ',end='')
                visited[start] = True
            
            for i in self.adj[start]:
                if (not visited[i]):
                    stack.append(i)

if __name__ == '__main__':
    adj = Graph(5)
    adj.addEdge(0, 1)
    adj.addEdge(0, 2)
    adj.addEdge(0, 3)
    adj.addEdge(1, 2)
    adj.addEdge(3, 4)
    adj.printGraph()
    adj.DFS(0)